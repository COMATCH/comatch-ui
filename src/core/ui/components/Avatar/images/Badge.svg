<?xml version="1.0" encoding="UTF-8"?>
<svg width="20px" height="24px" viewBox="0 0 20 24" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 53.2 (72643) - https://sketchapp.com -->
    <title>E7013BA0-2C2D-4043-A2FC-9860EC8294A5</title>
    <desc>Created with sketchtool.</desc>
    <g id="UI" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="project_details" transform="translate(-230.000000, -604.000000)" fill-rule="nonzero">
            <g id="card4" transform="translate(133.000000, 579.000000)">
                <g id="silver_status" transform="translate(97.000000, 25.000000)">
                    <path d="M13.135828,11.252948 L19.1724731,0.709595376 C19.1722907,0.521214988 19.097872,0.340623225 18.9655882,0.207548259 C18.8333045,0.0744732922 18.6539917,-0.000184055049 18.4670968,-2.85495643e-13 L0.746666667,-2.85495643e-13 C0.559533238,-0.00018429008 0.380011753,0.0746634145 0.247688373,0.208038498 C0.115364993,0.341413581 0.0411074857,0.522361898 0.0412903226,0.710982659 L6.70623656,10.8166474 L6.74890323,10.8811561 L13.135828,11.252948 Z" id="Shape" fill="#82B3BB"></path>
                    <path d="M6.67870968,10.8617341 L6.70623656,10.8166474 L13.488172,-3.39514372e-07 L0.746666667,-2.85495643e-13 C0.559533238,-0.00018429008 0.380011753,0.0746634145 0.247688373,0.208038498 C0.115364993,0.341413581 0.0411074857,0.522361898 0.0412903226,0.710982659 L6.70623656,10.8166474 L6.74890323,10.8811561" id="Shape" fill="#05677A"></path>
                    <ellipse id="Oval" fill="#C0C0C0" cx="9.61169892" cy="17.1024277" rx="6.78606452" ry="6.84"></ellipse>
                    <path d="M13.3539785,16.1986127 C13.3445193,16.282996 13.3024949,16.3602378 13.2369892,16.4136416 L11.603957,17.9972254 L11.9907097,20.2342197 C11.9945625,20.2638801 11.9961728,20.2937929 11.9955269,20.3236994 C11.9974259,20.3804655 11.9807398,20.4362842 11.948043,20.4825434 C11.9162927,20.5266282 11.864344,20.5512382 11.8104086,20.5477457 C11.7468681,20.5458888 11.6849278,20.5272308 11.6307957,20.4936416 L9.61101075,19.4379191 L7.59122581,20.4936416 C7.53670869,20.5263557 7.47499895,20.5449443 7.4116129,20.5477457 C7.3564475,20.5516417 7.30308127,20.5270966 7.26984946,20.4825434 C7.23837003,20.4358559 7.22290283,20.3800725 7.22580645,20.3236994 C7.22686506,20.2937092 7.22985342,20.2638198 7.23475269,20.2342197 L7.62150538,17.9972254 L5.9815914,16.4136416 C5.91903881,16.3584666 5.87906053,16.28183 5.86941935,16.1986127 C5.86941935,16.0880925 5.95337634,16.0187283 6.12129032,15.9905202 L8.3791828,15.663815 L9.39148387,13.6279769 C9.44791398,13.505896 9.52154839,13.4441618 9.61169892,13.4441618 C9.70184946,13.4441618 9.7747957,13.505896 9.83191398,13.6279769 L10.8442151,15.663815 L13.1021075,15.9905202 C13.2700215,16.0187283 13.3539785,16.0880925 13.3539785,16.1986127 L13.3539785,16.1986127 Z" id="Shape" fill="#FFFFFF"></path>
                </g>
            </g>
        </g>
    </g>
</svg>
