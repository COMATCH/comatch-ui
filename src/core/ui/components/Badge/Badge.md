
## Badge

From [`Badge.js`](Badge.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` |  | :x: | Additional class names
**color** | `Enum('green','orange','gray','white','blue','lightblue','darkGray')` | `'white'` | :x: | Text color of `Badge`
**icon** | `ReactNode` | `null` | :x: | SVG icon
**size** | `Enum('xs','sm')` | `'sm'` | :x: | 
**text** | `Union<String \| ReactNode>` | `''` | :x: | Text content of `Badge`



