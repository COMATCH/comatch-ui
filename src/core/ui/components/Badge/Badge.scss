@import '../../../styles/variables.scss';

.Badge {
    border-radius: 3px;
    border: 1px solid;
    display: inline-block;
    padding: 5px 8px;

    svg {
        margin-right: 5px;
        cursor: pointer;
    }

    // Size contextual classes
    &.xs {
        font-size: $co-font-xs;
        font-weight: bold;
        line-height: 1;
        padding: 2px 5px 3px 5px;

        svg {
            height: 10px;
        }
    }

    &.sm {
        font-size: $co-font-sm;

        svg {
            height: 12px;
        }
    }

    // Color contextual classes
    &:not(.white) {
        color: $co-color-white;
    }

    &.green {
        background-color: $co-color-green;
        border-color: $co-color-green;
    }

    &.orange {
        background-color: $co-color-orange;
        border-color: $co-color-orange;
    }

    &.gray {
        background-color: $co-color-gray;
        border-color: $co-color-gray;
    }

    &.white {
        background-color: $co-color-white;
        border-color: $co-color-gray;
        color: $co-color-gray;
    }

    &.blue {
        background-color: $co-color-secondary;
        border-color: $co-color-secondary;
    }

    &.lightblue {
        background-color: $co-color-primary;
        border-color: $co-color-primary;
    }
}
