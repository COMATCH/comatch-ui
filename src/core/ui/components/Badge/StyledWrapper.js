import React from 'react';
import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

export const getStylesBasedOnSize = ({ size }) => {
    switch (size) {
        case 'xs':
            return `
                border-radius: 9.5px;
                font-size: ${typography.fontXs};
                font-weight: bold;
                line-height: 1;
                padding: 3px 7px 5px;

                svg {
                    height: 10px;
                }
            `;

        case 'sm':
            return `
                border-radius: 15px;
                font-size: ${typography.fontSm};

                svg {
                    height: 12px;
                }
            `;

        default:
            return '';
    }
};

export const getColorBasedOnProps = ({ color }) => {
    switch (color) {
        case 'green':
            return `
                background-color: ${palette.green};
                border-color: ${palette.green};
                color: ${palette.white};
            `;

        case 'orange':
            return `
                background-color: ${palette.orange};
                border-color: ${palette.orange};
                color: ${palette.white};
            `;

        case 'gray':
            return `
                background-color: ${palette.gray};
                border-color: ${palette.gray};
                color: ${palette.white};
            `;

        case 'darkGray':
            return `
                background-color: ${palette.darkGray};
                border-color: ${palette.darkGray};
                color: ${palette.white};
            `;

        case 'white':
            return `
                background-color: ${palette.white};
                border-color: ${palette.gray};
                color: ${palette.gray};
            `;

        case 'blue':
            return `
                background-color: ${palette.secondary};
                border-color: ${palette.secondary};
                color: ${palette.white};
            `;

        case 'lightblue':
            return `
                background-color: ${palette.primary};
                border-color: ${palette.primary};
                color: ${palette.white};
            `;

        default:
            return '';
    }
};

export const StyledWrapper = styled(({ size, color, ...rest }) => <div {...rest} />)`
    border: 1px solid;
    display: inline-block;
    padding: 5px 8px;

    svg {
        margin-right: 5px;
        cursor: pointer;
    }

    ${getColorBasedOnProps}
    ${getStylesBasedOnSize}
`;
