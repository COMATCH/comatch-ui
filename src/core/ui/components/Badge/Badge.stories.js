/* eslint-disable import/no-unresolved */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { Badge } from './Badge';

storiesOf('Badge', module)
    .add('with xs size and lightblue background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="lightblue" />
            <code>{'<Badge text="Badge text" size="xs" color="lightblue" />'}</code>
        </>
    ))
    .add('with xs size and blue background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="blue" />
            <code>{'<Badge text="Badge text" size="xs" color="blue" />'}</code>
        </>
    ))
    .add('with xs size and orange background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="orange" />
            <code>{'<Badge text="Badge text" size="xs" color="orange" />'}</code>
        </>
    ))
    .add('with xs size and green background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="green" />
            <code>{'<Badge text="Badge text" size="xs" color="green" />'}</code>
        </>
    ))
    .add('with xs size and gray background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="gray" />
            <code>{'<Badge text="Badge text" size="xs" color="gray" />'}</code>
        </>
    ))
    .add('with xs size and white background-color', () => (
        <>
            <Badge text="Badge text" size="xs" color="white" />
            <code>{'<Badge text="Badge text" size="xs" color="white" />'}</code>
        </>
    ))
    .add('with sm size and lightblue background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="lightblue" />
            <code>{'<Badge text="Badge text" size="sm" color="lightblue" />'}</code>
        </>
    ))
    .add('with sm size and blue background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="blue" />
            <code>{'<Badge text="Badge text" size="sm" color="blue" />'}</code>
        </>
    ))
    .add('with sm size and orange background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="orange" />
            <code>{'<Badge text="Badge text" size="sm" color="orange" />'}</code>
        </>
    ))
    .add('with sm size and green background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="green" />
            <code>{'<Badge text="Badge text" size="sm" color="green" />'}</code>
        </>
    ))
    .add('with sm size and gray background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="gray" />
            <code>{'<Badge text="Badge text" size="sm" color="gray" />'}</code>
        </>
    ))
    .add('with sm size and white background-color', () => (
        <>
            <Badge text="Badge text" size="sm" color="white" />
            <code>{'<Badge text="Badge text" size="sm" color="white" />'}</code>
        </>
    ));
