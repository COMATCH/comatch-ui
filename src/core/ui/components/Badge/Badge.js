import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { StyledWrapper } from './StyledWrapper';
// import './Badge.scss';

export const allowedColors = ['green', 'orange', 'gray', 'white', 'blue', 'lightblue', 'darkGray'];

const propTypes = {
    /**
     * Additional class names
     */
    className: PropTypes.string,
    /**
     * Text color of `Badge`
     */
    color: PropTypes.oneOf(allowedColors),
    /**
     * SVG icon
     */
    icon: PropTypes.node,
    size: PropTypes.oneOf(['xs', 'sm']),
    /**
     * Text content of `Badge`
     */
    text: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};

const defaultProps = {
    color: 'white',
    icon: null,
    size: 'sm',
    text: '',
};

export const Badge = ({ className, color, icon, size, text }) => {
    const classes = classNames('Badge', className, color, size);
    return (
        <StyledWrapper className={classes} color={color} size={size}>
            {icon}
            {text}
        </StyledWrapper>
    );
};

Badge.defaultProps = defaultProps;

Badge.propTypes = propTypes;
