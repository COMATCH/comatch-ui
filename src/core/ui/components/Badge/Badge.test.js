/* eslint-env mocha */

import React from 'react';
import { shallow } from 'enzyme';
import { expect } from 'chai';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faDownload } from '@fortawesome/free-solid-svg-icons';
import { lorem } from 'faker';
import sample from 'lodash/sample';
import { allowedColors, Badge } from './Badge';
import '../../../config/tests/setup';

describe('Badge', () => {
    it('should render a Badge with default props', () => {
        const item = shallow(<Badge />);
        expect(item.hasClass('Badge')).to.equal(true);
        expect(item.hasClass('white')).to.equal(true);
        expect(item.hasClass('sm')).to.equal(true);
        expect(item.find(FontAwesomeIcon).length).to.equal(0);
        expect(item.text()).to.equal('');
    });
    it('should render a Badge with a given color', () => {
        const color = sample(allowedColors);
        const item = shallow(<Badge color={color} />);
        expect(item.hasClass(color)).to.equal(true);
    });
    it('should render a Badge with a given size', () => {
        const item = shallow(<Badge size="xs" />);
        expect(item.hasClass('xs')).to.equal(true);
    });
    it('should render a Badge with a given icon', () => {
        const icon = <FontAwesomeIcon icon={faDownload} />;
        const item = shallow(<Badge icon={icon} />);
        expect(item.find(FontAwesomeIcon).length).to.equal(1);
    });
    it('should render a Badge with given text', () => {
        const text = lorem.word();
        const item = shallow(<Badge text={text} />);
        expect(item.text()).to.equal(text);
    });
});
