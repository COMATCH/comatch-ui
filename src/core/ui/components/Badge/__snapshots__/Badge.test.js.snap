// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly 1`] = `
<div
  className="Badge white sm"
>
  
</div>
`;
